rct229/rulesets/ashrae9012019/section19/section19rule15.py

```diff
@@ -0,0 +1,213 @@
+                        terminal_p, "Terminal", "primary_airflow"
+                    )
+
+        return {
+            "baseline_system_types_dict": baseline_system_types_dict,
+            "hvac_info_dict_b": hvac_info_dict_b,
+        }
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+
+        return any(
+            [
+                baseline_system_type_compare(system_type, applicable_sys_type, False)
+                for system_type in baseline_system_types_dict.keys()
+                for applicable_sys_type in APPLICABLE_SYS_TYPES
+            ]
+        )
+
+    class HVACRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section19Rule15.HVACRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                required_fields={
+                    "$": ["fan_system"],
+                    "fan_system": ["minimum_outdoor_airflow"],
+                },
+            )
+
+        def is_applicable(self, context, data=None):
+            hvac_b = context.baseline
+            hvac_id_b = hvac_b["id"]
+            baseline_system_types_dict = data["baseline_system_types_dict"]
+
+            return any(
+                hvac_id_b in baseline_system_types_dict[system_type]
+                for system_type in baseline_system_types_dict.keys()
+            )
+
+        def get_calc_vals(self, context, data=None):
+            hvac_b = context.baseline
+            hvac_id_b = hvac_b["id"]
+
+            fan_system_b = hvac_b["fan_system"]
+
+            fan_info_b = (
+                get_fan_system_object_supply_return_exhaust_relief_total_power_flow(
+                    fan_system_b
+                )
+            )
+
+            supply_fan_qty_b = fan_info_b["supply_fans_qty"]
+            supply_fan_airflow_b = fan_info_b["supply_fans_airflow"]
+
+            minimum_outdoor_airflow_b = fan_system_b["minimum_outdoor_airflow"]
+
+            return {
+                "hvac_id_b": hvac_id_b,
+                "supply_fan_qty_b": supply_fan_qty_b,
+                "supply_fan_airflow_b": supply_fan_airflow_b,
+                "minimum_outdoor_airflow_b": minimum_outdoor_airflow_b,
+            }
+
+        def manual_check_required(self, context, calc_vals=None, data=None):
+            hvac_id_b = calc_vals["hvac_id_b"]
+            supply_fan_qty_b = calc_vals["supply_fan_qty_b"]
+            supply_fan_airflow_b = calc_vals["supply_fan_airflow_b"]
+            hvac_info_dict_b = data["hvac_info_dict_b"][hvac_id_b]
+
+            return (
+                supply_fan_qty_b == 1
+                and not hvac_info_dict_b["all_design_setpoints_105"]
+                and std_equal(
+                    hvac_info_dict_b["proposed_supply_cfm"],
+                    supply_fan_airflow_b,
+                )
+            ) and (supply_fan_qty_b != 1)
```

Comment:
Here shall be `or`.
According to the RDS logic, this line should be `supply_fan_qty_b > 1`

However, in that case, it does not cover cases `<1` - most likely is `supply_fan_qty_b = 0`

We can discuss about that situation but in here, you should either update it to `supply_fan_qty_b > 1` or update the manual check message to include the scenario 0.

Reply:
I changed the undetermined message. Addressed.